### ISSUE TITLE
Missing timestamp

### ISSUE THREAD

**Issue by rvanspaa** [SDL-Hercules-390/hyperion#259]
Hi,

when running Hyperion V4.2.1 (compiled without specifying any special options) under Linux 2.6.32-754.22.1.el6.x86_64, whether run in daemon mode or not, no timestamp is ever added to log messages. E.g. when not in daemon mode:
```
HHC01603I logopt time
HHC02204I LOGOPT         set to TIMESTAMP
HHC01603I logopt notime
HHC02204I LOGOPT         set to NOTIMESTAMP
HHC01603I logopt timestamp
HHC02204I LOGOPT         set to TIMESTAMP
HHC01603I msglevel
HHC17012I MSGLEVEL = verbose nodebug emsgloc
 IWM063I WLM POLICY WAS REFRESHED DUE TO A PROCESSOR SPEED CHANGE
herc =====>
CP00 PSW=0706000000000000 0000000000000000 24..W....Z                                                                                           instcnt 21,497,120,612; mips 7.248; I/O      8
```
PS when running V3.11 in daemon mode, there is also no timestamp.

**Comment by Fish-Git** [comment]
> ... whether run in daemon mode or not, no timestamp is ever added to log messages. E.g. when not in daemon mode: ...

Timestamps currently _never_ appear on panel messages, only on logfile messages, under the control of the [`LOGOPT`](https://sdl-hercules-390.github.io/html/hercconf.html#LOGOPT) statement/command.

That having been said, I did notice a bug wherein the timestamp is _not_ being honored when in daemon mode, even though the `logopt` option was indeed set to `timestamp`.  This will be fixed shortly. (The `logopt` option _is_ being correctly honored in normal _(non-daemon)_ mode however.)

As far as timestamping messages that appear on the _panel_ goes, as I said, _not_ timestamping them is normal behavior (i.e. as designed).

_However, **that** having been said,_ timestamping panel messages too _is_ one of our long term goals (it's on our "todo" list), but we simply haven't found the "time" (heh!) yet to work on it. (It's not exactly a priority item.)

**Comment by rvanspaa** [comment]
Hi,

When I run in panel mode in the foreground, with the log command in the rc file,
then the log has a timestamp.
If I run with nohup and in background mode, but without specifying daemon mode,
and still using the log command, then the log has no timestamp.

I need to run hercules continuously, and with a timestamp in the log, without
being continuously logged on.

I await your fix for daemon mode with anticipation, and would much appreciate it
if your could update this thread when it's available.

BTW if you are going to fix the bug, it would also be much appreciated if you
could add the (short) date (using the default date format for the underlying
operating system) to the timestamp while you were at it. If the format is too
difficult to obtain in all cases, then perhaps just use a fixed format, and
publish it?

**Comment by Fish-Git** [comment]
@rvanspaa Please do not reply to GitHub via email. I know it's convenient, but it makes it impossible for us to make minor edits to your reply (for clarity / formatting purposes). Please reply only via the GitHub webpage user interface. Thanks.

**Comment by rvanspaa** [comment]
...then perhaps someone should remove that option from the notification emails that are sent out?
I'm new to this list, and that was my first reply, so I took the notification email as my guide.

**Comment by Fish-Git** [comment]
> Then perhaps someone should remove that option from the notification emails that are sent out?

I would if I could!

And I must apologize. It was slightly rude of me to ask you to please reply via the GitHub webpage user interface. Rather, I should have simply notified you that I/we would _prefer_ that you do so.

Asking someone to please do something seems like too much of a demand (regardless of the fact that it was prefaced with "Please"). Simply informing someone of a preferred behavior on the other hand is more polite (less demanding) IMO, leaving it up to the other party to decide whether or not they care enough about the other and their preference to make the effort to comply.&nbsp;`:)`

But thanks for doing so.&nbsp;`:)`

Now... about this issue... you said:

> If I run with nohup and in background mode ...

I don't know what that means. What is "nohup"?

> ... and still using the log command ...

Is there a reason you are using the `log` command rather than simply redirecting output to a log file? E.g. `hercules -f my.conf > mylog.txt`

> I need to run hercules continuously, and with a timestamp in the log, without being continuously logged on.

I understand perfectly. That's what the background "daemon mode" was meant for.

> BTW if you are going to fix the bug, it would also be much appreciated if you could add the (short) date (using the default date format for the underlying operating system) to the timestamp while you were at it. If the format is too difficult to obtain in all cases, then perhaps just use a fixed format, and publish it?

Please create a new GitHub Issue for that so we can discuss it and track its progress. I prefer to keep each submitted GitHub Issue restricted to resolving one problem/issue/request at a time.

Thanks.

**Comment by Fish-Git** [comment]
Closed by commit a0d61fbcc11faf47998a1b165a964758cb83c606.

@rvanspaa: Please confirm that this adequately resolves your issue. Thank you.

**Comment by rvanspaa** [comment]
Hi,
Many thanks, the fix works as intended.
